### ISSUE TITLE
[QA] ownCloud becomes defunct, when server_password is incorrect

### ISSUE THREAD

**Issue by jnweiger** [owncloud/search_elastic#281]
Seen in 2.1.1-rc1

* Visit admin UI to set server, username and password.
* The connection becomes green, occ search:index:update works. OK.
* Inspect the password with `occ config:app:get search_elastic server_password` - it is encrypted. OK
* Try set a new password with: `occ config:app:set search_elastic server_password --value "newpassword"`
* Reload the web UI -> Error message: "Internal Server Error". BAD
* Inspect the password with `occ config:app:get search_elastic server_password` - the command fails. BAD.
   `Exception: Authenticated ciphertext could not be decoded. in /var/www/owncloud/lib/private/Security/Crypto.php:161`
* Inspect the password with `mysql owncloud -e 'select * from oc_appconfig where configkey = "server_password"' - The configvalue is unencrypted. BAD
* ownCloud is now defunct. the admin UI cannot be used to correctly set a password. BAD.
* occ is also defunct, and cannot fix the situation. BAD.

Expected behaviour
* Incorrectly formatted encrpyted passwords are rejected. (or password can be specified unencrypted and is automatically encrypted in this case - but that would be specific to search-elastic then ...)
* An incorrectly formatted encrypted password should only disable the elastic_search connection.
* The rest of owncloud should continue working. 

Excerpt from owncloud.log:

```
{"reqId":"YmKpxl2hGa0afP1E7CutGgAAAAs","level":0,"time":"2022-04-22T13:12:38+00:00","remoteAddr":"45.155.204.146","user":"--","app":"OC\\Authentication\\Token\\DefaultTokenProvider::invalidateToken","method":"GET","url":"\/","message":"invalidating token 7fc332a9bbf6d8bdbfef82b3958357bd24626454cfcbe84070e13d9e1067c3087ebfb8673efa85323b1df2c40d56f4cad04964529f82aa14b3434997ab979c34"}
{"reqId":"YmKpxl2hGa0afP1E7CutGgAAAAs","level":3,"time":"2022-04-22T13:12:38+00:00","remoteAddr":"45.155.204.146","user":"--","app":"no app in context","method":"GET","url":"\/","message":"Exception: {\"Exception\":\"Exception\",\"Message\":\"Authenticated ciphertext could not be decoded.\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/apps-external\\\/search_elastic\\\/lib\\\/SearchElasticConfigService.php(134): OC\\\\Security\\\\Crypto->decrypt(*** sensitive parameters replaced ***)\\n#1 \\\/var\\\/www\\\/owncloud\\\/apps-external\\\/search_elastic\\\/lib\\\/SearchElasticConfigService.php(258): OCA\\\\Search_Elastic\\\\SearchElasticConfigService->getServerPassword()\\n#2 \\\/var\\\/www\\\/owncloud\\\/apps-external\\\/search_elastic\\\/lib\\\/SearchElasticConfigService.php(142): OCA\\\\Search_Elastic\\\\SearchElasticConfigService->parseServers()\\n#3 \\\/var\\\/www\\\/owncloud\\\/apps-external\\\/search_elastic\\\/lib\\\/AppInfo\\\/Application.php(76): OCA\\\\Search_Elastic\\\\SearchElasticConfigService->getParsedServers()\\n#4 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/pimple\\\/pimple\\\/src\\\/Pimple\\\/Container.php(122): OCA\\\\Search_Elastic\\\\AppInfo\\\\Application->OCA\\\\Search_Elastic\\\\AppInfo\\\\{closure}(*** sensitive parameters replaced ***)\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Utility\\\/SimpleContainer.php(108): Pimple\\\\Container->offsetGet()\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Utility\\\/SimpleContainer.php(64): OC\\\\AppFramework\\\\Utility\\\\SimpleContainer->query()\\n#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Utility\\\/SimpleContainer.php(90): OC\\\\AppFramework\\\\Utility\\\\SimpleContainer->buildClass()\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Utility\\\/SimpleContainer.php(110): OC\\\\AppFramework\\\\Utility\\\\SimpleContainer->resolve()\\n#9 \\\/var\\\/www\\\/owncloud\\\/apps-external\\\/search_elastic\\\/lib\\\/AppInfo\\\/Application.php(102): OC\\\\AppFramework\\\\Utility\\\\SimpleContainer->query()\\n#10 \\\/var\\\/www\\\/owncloud\\\/apps-external\\\/search_elastic\\\/appinfo\\\/app.php(30): OCA\\\\Search_Elastic\\\\AppInfo\\\\Application->init()\\n#11 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(253): require_once('\\\/var\\\/www\\\/ownclo...')\\n#12 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(192): OC_App::requireAppFile()\\n#13 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(125): OC_App::loadApp()\\n#14 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/util.php(150): OC_App::loadApps()\\n#15 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Files\\\/Filesystem.php(264): OC_Util::setupFS()\\n#16 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Files\\\/View.php(1449): OC\\\\Files\\\\Filesystem::getMountManager()\\n#17 \\\/var\\\/www\\\/owncloud\\\/apps\\\/files_versions\\\/lib\\\/MetaStorage.php(80): OC\\\\Files\\\\View->getFileInfo()\\n#18 \\\/var\\\/www\\\/owncloud\\\/apps\\\/files_versions\\\/lib\\\/AppInfo\\\/Application.php(71): OCA\\\\Files_Versions\\\\MetaStorage->__construct()\\n#19 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/pimple\\\/pimple\\\/src\\\/Pimple\\\/Container.php(122): OCA\\\\Files_Versions\\\\AppInfo\\\\Application->OCA\\\\Files_Versions\\\\AppInfo\\\\{closure}(*** sensitive parameters replaced ***)\\n#20 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Utility\\\/SimpleContainer.php(108): Pimple\\\\Container->offsetGet()\\n#21 \\\/var\\\/www\\\/owncloud\\\/apps\\\/files_versions\\\/lib\\\/AppInfo\\\/Application.php(76): OC\\\\AppFramework\\\\Utility\\\\SimpleContainer->query()\\n#22 \\\/var\\\/www\\\/owncloud\\\/apps\\\/files_versions\\\/appinfo\\\/app.php(28): OCA\\\\Files_Versions\\\\AppInfo\\\\Application->__construct()\\n#23 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(253): require_once('\\\/var\\\/www\\\/ownclo...')\\n#24 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(192): OC_App::requireAppFile()\\n#25 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(125): OC_App::loadApp()\\n#26 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(915): OC_App::loadApps()\\n#27 \\\/var\\\/www\\\/owncloud\\\/index.php(54): OC::handleRequest()\\n#28 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Security\\\/Crypto.php\",\"Line\":161}"}
{"reqId":"YmKpxl2hGa0afP1E7CutGgAAAAs","level":2,"time":"2022-04-22T13:12:38+00:00","remoteAddr":"45.155.204.146","user":"--","app":"core","method":"GET","url":"\/","message":"Could not load app \"search_elastic\", see exception above"}
{"reqId":"YmKpxl2hGa0afP1E7CutGgAAAAs","level":3,"time":"2022-04-22T13:12:38+00:00","remoteAddr":"45.155.204.146","user":"--","app":"no app in context","method":"GET","url":"\/","message":"Exception: {\"Exception\":\"Exception\",\"Message\":\"Authenticated ciphertext could not be decoded.\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/apps-external\\\/search_elastic\\\/lib\\\/SearchElasticConfigService.php(134): OC\\\\Security\\\\Crypto->decrypt(*** sensitive parameters replaced ***)\\n#1 \\\/var\\\/www\\\/owncloud\\\/apps-external\\\/search_elastic\\\/lib\\\/SearchElasticConfigService.php(258): OCA\\\\Search_Elastic\\\\SearchElasticConfigService->getServerPassword()\\n#2 \\\/var\\\/www\\\/owncloud\\\/apps-external\\\/search_elastic\\\/lib\\\/SearchElasticConfigService.php(142): OCA\\\\Search_Elastic\\\\SearchElasticConfigService->parseServers()\\n#3 \\\/var\\\/www\\\/owncloud\\\/apps-external\\\/search_elastic\\\/lib\\\/AppInfo\\\/Application.php(76): OCA\\\\Search_Elastic\\\\SearchElasticConfigService->getParsedServers()\\n#4 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/pimple\\\/pimple\\\/src\\\/Pimple\\\/Container.php(122): OCA\\\\Search_Elastic\\\\AppInfo\\\\Application->OCA\\\\Search_Elastic\\\\AppInfo\\\\{closure}(*** sensitive parameters replaced ***)\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Utility\\\/SimpleContainer.php(108): Pimple\\\\Container->offsetGet()\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Utility\\\/SimpleContainer.php(64): OC\\\\AppFramework\\\\Utility\\\\SimpleContainer->query()\\n#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Utility\\\/SimpleContainer.php(90): OC\\\\AppFramework\\\\Utility\\\\SimpleContainer->buildClass()\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Utility\\\/SimpleContainer.php(110): OC\\\\AppFramework\\\\Utility\\\\SimpleContainer->resolve()\\n#9 \\\/var\\\/www\\\/owncloud\\\/apps-external\\\/search_elastic\\\/lib\\\/AppInfo\\\/Application.php(102): OC\\\\AppFramework\\\\Utility\\\\SimpleContainer->query()\\n#10 \\\/var\\\/www\\\/owncloud\\\/apps-external\\\/search_elastic\\\/appinfo\\\/app.php(30): OCA\\\\Search_Elastic\\\\AppInfo\\\\Application->init()\\n#11 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(253): require_once('\\\/var\\\/www\\\/ownclo...')\\n#12 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(192): OC_App::requireAppFile()\\n#13 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(125): OC_App::loadApp()\\n#14 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/util.php(150): OC_App::loadApps()\\n#15 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Files\\\/Filesystem.php(264): OC_Util::setupFS()\\n#16 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Files\\\/View.php(1449): OC\\\\Files\\\\Filesystem::getMountManager()\\n#17 \\\/var\\\/www\\\/owncloud\\\/apps\\\/files_versions\\\/lib\\\/MetaStorage.php(80): OC\\\\Files\\\\View->getFileInfo()\\n#18 \\\/var\\\/www\\\/owncloud\\\/apps\\\/files_versions\\\/lib\\\/AppInfo\\\/Application.php(71): OCA\\\\Files_Versions\\\\MetaStorage->__construct()\\n#19 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/pimple\\\/pimple\\\/src\\\/Pimple\\\/Container.php(122): OCA\\\\Files_Versions\\\\AppInfo\\\\Application->OCA\\\\Files_Versions\\\\AppInfo\\\\{closure}(*** sensitive parameters replaced ***)\\n#20 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Utility\\\/SimpleContainer.php(108): Pimple\\\\Container->offsetGet()\\n#21 \\\/var\\\/www\\\/owncloud\\\/apps\\\/files_versions\\\/lib\\\/AppInfo\\\/Application.php(76): OC\\\\AppFramework\\\\Utility\\\\SimpleContainer->query()\\n#22 \\\/var\\\/www\\\/owncloud\\\/apps\\\/files_versions\\\/appinfo\\\/app.php(28): OCA\\\\Files_Versions\\\\AppInfo\\\\Application->__construct()\\n#23 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(253): require_once('\\\/var\\\/www\\\/ownclo...')\\n#24 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(192): OC_App::requireAppFile()\\n#25 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(125): OC_App::loadApp()\\n#26 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(915): OC_App::loadApps()\\n#27 \\\/var\\\/www\\\/owncloud\\\/index.php(54): OC::handleRequest()\\n#28 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Security\\\/Crypto.php\",\"Line\":161}"}
{"reqId":"YmKpxl2hGa0afP1E7CutGgAAAAs","level":2,"time":"2022-04-22T13:12:38+00:00","remoteAddr":"45.155.204.146","user":"--","app":"core","method":"GET","url":"\/","message":"Could not load app \"files_versions\", see exception above"}
...
```

**Comment by ahherrera** [comment]
Hi @jnweiger !!
I have been testing with the master branch and the 2.1.1.1-rc1 and both have changed the password fine from the command line. 
We can schedule a call and together we can look into the error.
![Captura de pantalla 2022-04-21 195644](https://user-images.githubusercontent.com/6892751/164577588-2a8d2b29-f8ff-4d06-85fc-7b1f58e50ddc.png)

**Comment by ahherrera** [comment]
The error is related to the core and not to the elastic search app. The bug is fixed and security controls are put in place when displaying the password value through the command line.
![Captura de pantalla 2022-04-24 165452](https://user-images.githubusercontent.com/6892751/164998806-b52f1577-7c7e-4057-835e-cb1259f17e6e.png)

**Comment by jnweiger** [comment]
Protection against non-decodable database values should be done inside the app.

**Comment by jvillafanez** [comment]
Is it reproducible with the latest code? The password is now stored in the oc_credentials table, and it's fetched from there. Setting the value from the `config:app:set` command shouldn't matter

**Comment by jnweiger** [comment]
Ah, that is good improvement. I'll give it a try. Thank you!

**Comment by jnweiger** [comment]
Confirmed fixed in 2.2.0-RC1
